frontend/app/utils/FutureSupplier.scala

```diff
@@ -0,0 +1,19 @@
+package utils
+
+import akka.agent.Agent
+
+import scala.concurrent.Future
+import play.api.libs.concurrent.Execution.Implicits._
+
+class FutureSupplier[T](f: => Future[T]) {
+  private val agent = Agent(f)
+
+  def refresh(): Future[T] = for {
+    refreshAlteringFuture <- agent.alter { currentRefresh =>
+      if (currentRefresh.isCompleted) f else currentRefresh
```

Comment:
can we avoid reassigning the same value back to the agent, as it might cause an additional blocking of the agent while this happens, which is slightly suboptiomal?